### ISSUE TITLE
Less autocompletion not working as expected

### ISSUE THREAD

**Issue by babobski** [Komodo/KomodoEdit#1583]
### Short Summary
Removed the less autocompletion form my less addon, only the native less support is not working as expected. It doesn't include imports. Its only picking up the vars declared in the current document.

### Steps to Reproduce
Create a file main.less:
```less
@black: #333;
@white: #fff;

@import 'less/menu.less';
```

Now in the less/menu.less 
```
@grey: #ccc;

.menu {
	color: 
}
```

When you now type a `@` you only get autocompletion for green and not for black and white.

### Expected results
You would get autocompletion for green, black and white

### Actual results
Only the vars in the current document shows up in the autocompletion

### Platform Information
Komodo 10 IDE

**Comment by mitchell-as** [comment]
I wasn't aware you could import variables from other Less files. This feature simply doesn't exist :) Thanks for letting me know.

**Comment by Defman21** [comment]
Note @mitchell-as this feature exists in SCSS as well.

**Comment by Defman21** [comment]
Also using Middleman, in SCSS, I can import these files without extension if the name of a file starts with _, like "_colors.scss" and in my `index.css.scss` file:
```scss
@import "colors";
```

**Comment by Naatan** [comment]
@mitchell-as is this trivial enough to be marked for 10.1?

**Comment by mitchell-as** [comment]
I'm not sure. I'd like to have it marked for 10.1, but some investigation may push it back.

**Comment by babobski** [comment]
I think it's a pretty tough one.
If you have a basic example like this:

```
/less
-- main.less
-- menu.less
-- respond.less
default.less
```

The imports are in the default.less.
When I open the project and I want to edit the main.less, I need to get the less variables from all the files.
Less also got something called [lazy loading](http://lesscss.org/features/#variables-feature-lazy-loading), i can use a variable in my first imported file that i declare in the last less file, this is still valid less.

If you got a more complex structure, lets say I have a vendor with multiple packages:

```
/package01
--/css
---- style.less
----/less
------ global.less
------ win.less
/package02
-- style.less
-- /less
---- global.less
---- main.less
```

When you edit both global less files, you don't want the variables to get mixed.
In both files you only want all the variables of the belonging package.

As @Defman21 stated, there are some things you have to look out for when processing less and sass imports.

Valid imports in less:
```less
@import "test/main";
@import "test/main.less";

@path: "test";
@import: "@path/main.less";

@import (reference) "test/main.less";
@import (inline, optional) "test/main.less";
```

And for scss:
```scss
@import 'test/main';
@import 'test/_main';
@import 'test/main.scss'; // I think this is still valid
```

And for sass:
```sass
@import test/main
@import test/_main
@import test/main.sass // I think this is still valid
```

The sass docs state this about adding `.scss`:
> Notice we're using @import 'reset'; in the base.scss file. When you import a file you don't need to include the file extension .scss. Sass is smart and will figure it out for you.

**Comment by Naatan** [comment]
Yeah I know less imports in particular are really really crafty. Honestly this may just have to be "fuzzy" by design.

**Comment by babobski** [comment]
yes, less is pretty complicated, it's got all kind of function, I think only a few will use.

**Comment by mitchell-as** [comment]
Note to self: It sounds like this should be possible after the scheduled architecture changes and "guessing" queries.

**Comment by Defman21** [comment]
Note 2: Sass/SCSS uses `$` for variables, Less uses `@`
